OpenRA.Mods.Cnc/Traits/Buildings/Occupying.cs

```diff
@@ -0,0 +1,64 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Cnc.Traits
+{
+	public class OccupyingInfo : ITraitInfo
+	{
+		[Desc("The speech notification to play to the old owner.")]
+		public readonly string LoadNotification = null;
+		public readonly string UnloadNotification = null;
+
+		public object Create(ActorInitializer init) { return new Occupying(init.Self, this); }
+	}
+
+	public class Occupying : ICargoLoaded, ICargoUnloaded
+	{
+		readonly OccupyingInfo info;
+		readonly Actor self;
+		public Occupying(Actor self, OccupyingInfo info)
+		{
+			this.info = info;
+			this.self = self;
+		}
+
+		public void CargoLoaded(Actor actor)
+		{
+			var newOwner = actor.Owner;
+			if (self.Owner == newOwner)
+				return;
+			ChangeOwner(actor, newOwner);
+
+			Game.Sound.PlayNotification(self.World.Map.Rules, actor.Owner, "Speech", info.LoadNotification, newOwner.Faction.InternalName);
+		}
+
+		public void CargoUnloaded(Actor actor)
+		{
+			if (self.Trait<Cargo>().PassengerCount > 0)
+				return;
+
+			Game.Sound.PlayNotification(self.World.Map.Rules, actor.Owner, "Speech", info.UnloadNotification, actor.Owner.Faction.InternalName);
+			ChangeOwner(actor, self.World.WorldActor.Owner);
```

Comment:
I found INotifyPassengerEntered and decided use it instead ICargoLoaded, but trait not fire on initialization. It's right. Don't play sound for example.
If you make change owner cargo with passanger on init map, you don't initialize originalOwner.
Maybe change originalOwner on yaml file?